File: Java/Stack.java

class node 
{
    public int data;
    public node next;
    
};

class StackX 
{
    private node first;
    private int iCount;

    public StackX()
    {
       this.first = null;
       this.iCount = 0;
    }

    public void push(int no) //Insert First
    {
        node newn = null;
        newn = new node();

        newn.data = no;
        newn.next = null;
       
        if(first == null)
        {
            first = newn;
        }
        else
        {
            newn.next = first;
            first = newn;
        }

        iCount++;
    }

    public int pop()   //DeleteFirst
    {
        node temp = first;

        if(this.first == null)
        {
            System.out.println("Unbale to pop the element :");
            return -1;
        }
        else
        {
           first = first.next;
            iCount--;
            return temp.data;
        }
    }

    public void Display()
    {
        node temp = null;
        temp = first;

        while(temp != null)
        {
            System.out.println("| "+temp.data+" |");
            temp = temp.next;
        }
        // System.out.println("NUll");
    }

    public int Count()
    {
        return iCount;
    }


}
public class Stack
{
    public static void main (String args[])
    {
       StackX sobj = new StackX();
       int iRet = 0;

       sobj.push(11);
       sobj.push(12);
       sobj.push(13);
       sobj.push(14);
       sobj.push(15);

       sobj.pop();
       sobj.pop();

       sobj.Display();
       iRet = sobj.Count();
       System.out.println("Number of the Element in the stack :"+ iRet);
    }
}

File: CPP/Stack.cpp
#include<iostream>
using namespace std;

struct node 
{
    int data;
    struct node *next;
};

class StackX
{
private:
    struct node *first;
    int iCount;

public:
    StackX();        
    void push(int no);      
    int pop();               
    void Display(); 
    int Count();             
};

StackX::StackX()
{
    first = NULL;
    iCount = 0;
}

void StackX :: push(int no)
{
    struct node *newn = new node();

    newn-> data = no;
    newn -> next = NULL;

    // first = newn;

    newn -> next = first;
    first = newn;

    iCount++;
}

int StackX :: pop()
{
    struct node *temp = first;
    int value = 0;

    if(first == NULL)
    {
        cout << "Unable to pop the elmenet in stack:\n" << endl;
        return -1; 
    }
    else
    {
        temp = first;
        value = temp -> data;
        
        first  = first -> next;
        delete temp;

        iCount--;
        return value;
    }

}

void StackX::Display()
{
    struct node *temp = first;

    while (temp != NULL)
    {
        cout << "| " << temp->data << " |\n";
        temp = temp->next;
    }
    cout << "\n";
}

int StackX::Count()
{
    return iCount;
}

int main()
{
    StackX *sobj = new StackX();

    int iRet = 0;

    sobj->push(10);
    sobj->push(20);
    sobj->push(30);

    sobj->Display();

    iRet = sobj->Count();
    cout << "Number of elements in the stack: " << iRet << "\n";

    iRet = sobj->pop();
    cout << "Popped element is: " << iRet << "\n";

    iRet = sobj->pop();
    cout << "Popped element is: " << iRet << "\n";

    iRet = sobj->pop();
    cout << "Popped element is: " << iRet << "\n";

    sobj->Display();

    iRet = sobj->Count();
    cout << "Number of elements in the stack: " << iRet << "\n";

    delete sobj;

    return 0;
}